// ng2-materialize/dist/collapsible/collapsible-item/collapsible-item.component.js
import { Component, Directive, ElementRef, Input } from '@angular/core';
var MzCollapsibleItemComponent = (function () {
    function MzCollapsibleItemComponent(element) {
        this.element = element;
    }
    MzCollapsibleItemComponent.prototype.ngAfterViewInit = function () {
        this.innerHTML = this.element.nativeElement.innerHTML;
    };
    return MzCollapsibleItemComponent;
}());
export { MzCollapsibleItemComponent };
MzCollapsibleItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'mz-collapsible-item',
                template: "<div class=\"collapsible-header\" [class.active]=\"active\"><ng-content select=\"mz-collapsible-item-header\"></ng-content></div><div class=\"collapsible-body\"><ng-content select=\"mz-collapsible-item-body\"></ng-content></div>",
                styles: [""],
            },] },
];
/** @nocollapse */
MzCollapsibleItemComponent.ctorParameters = function () { return [
    { type: ElementRef, },
]; };
MzCollapsibleItemComponent.propDecorators = {
    'active': [{ type: Input },],
};
// Declare the tags to avoid error: '<mz-collapsible-item-x>' is not a known element
// https://github.com/angular/angular/issues/11251
// tslint:disable: directive-selector
var MzCollapsibleItemBodyDirective = (function () {
    function MzCollapsibleItemBodyDirective() {
    }
    return MzCollapsibleItemBodyDirective;
}());
export { MzCollapsibleItemBodyDirective };
MzCollapsibleItemBodyDirective.decorators = [
    { type: Directive, args: [{ selector: 'mz-collapsible-item-body' },] },
];
/** @nocollapse */
MzCollapsibleItemBodyDirective.ctorParameters = function () { return []; };
var MzCollapsibleItemHeaderDirective = (function () {
    function MzCollapsibleItemHeaderDirective() {
    }
    return MzCollapsibleItemHeaderDirective;
}());
export { MzCollapsibleItemHeaderDirective };
MzCollapsibleItemHeaderDirective.decorators = [
    { type: Directive, args: [{ selector: 'mz-collapsible-item-header' },] },
];
/** @nocollapse */
MzCollapsibleItemHeaderDirective.ctorParameters = function () { return []; };
