// ng2-materialize/inline-template/collapsible/collapsible-item/collapsible-item.component.ts
import {
  AfterViewInit,
  Component,
  Directive,
  ElementRef,
  Input } from '@angular/core';

@Component({
  selector: 'mz-collapsible-item',
  template: `<div class="collapsible-header" [class.active]="active"><ng-content select="mz-collapsible-item-header"></ng-content></div><div class="collapsible-body"><ng-content select="mz-collapsible-item-body"></ng-content></div>`,
  styles: [``],
})
export class MzCollapsibleItemComponent implements AfterViewInit {
  @Input() active: boolean;

  innerHTML: string;

  constructor(public element: ElementRef) { }

  ngAfterViewInit() {
    this.innerHTML = this.element.nativeElement.innerHTML;
  }
}

// Declare the tags to avoid error: '<mz-collapsible-item-x>' is not a known element
// https://github.com/angular/angular/issues/11251
// tslint:disable: directive-selector
@Directive({ selector: 'mz-collapsible-item-body' }) export class MzCollapsibleItemBodyDirective { }
@Directive({ selector: 'mz-collapsible-item-header' }) export class MzCollapsibleItemHeaderDirective { }
